client/js/widgets/widget.js

```diff
@@ -1025,12 +1025,31 @@ export class Widget extends StateManaged {
           for(const key in a.in)
             await callWithAdditionalValues({ key, value: a.in[key] }, {});
           if(jeRoutineLogging)
-            jeLoggingRoutineOperationSummary( `element in '${JSON.stringify(a.in)}'`);
+            jeLoggingRoutineOperationSummary( `elements in '${JSON.stringify(a.in)}'`);
+        } else if(a.range) {
+          let range = a.range;
+          if(!Array.isArray(range))
+            range = [1,range];
+          if(range.length < 2 || range[1]<0 ||(range[1] && range[1]<0)) {
```

Comment:
I'm not following the logic. Clearly range.length needs to be >= 2, but the tests for range[1] don't seem quite right.